File: API/PCPartPicker/Models/Videocard.cs
using PCPartPicker.Interfaces;
using System;
using System.Collections.Generic;

namespace PCPartPicker.Models;

public partial class Videocard : IComponent
{
    public int Id { get; set; }

    public string? Name { get; set; }

    public decimal? Price { get; set; }

    public string? Chipset { get; set; }

    public decimal? Memory { get; set; }

    public int? CoreClock { get; set; }

    public int? BoostClock { get; set; }

    public string? Color { get; set; }

    public int? Length { get; set; }

    public int? ManufacturerTypeId { get; set; }

    public int? ChipsetTypeId { get; set; }

    public int? SeriesTypeId { get; set; }

    public virtual ChipsetType? ChipsetType { get; set; }

    public virtual ICollection<Configuration> Configurations { get; set; } = new List<Configuration>();

    public virtual ManufacturerType? ManufacturerType { get; set; }

    public virtual SeriesType? SeriesType { get; set; }
}

File: UI/pc-part-picker/src/model/Videocard.ts
import { ChipsetType } from "./ChipsetType";
import { ManufacturerType } from "./ManufacturerType";
import { PcPart } from "./Pcpart";
import { SeriesType } from "./SeriesType";

export interface Videocard extends PcPart {
    chipset: string | null;
    memory: number | null;
    coreClock: number | null;
    boostClock: number | null;
    color: string | null;
    length: number | null;
    manufacturerTypeId: number | null;
    chipsetTypeId: number | null;
    seriesTypeId: number | null;
    chipsetType: string | null;
    manufacturerType: string | null;
    seriesType: string | null;
}